comms/src/message/envelope.rs

```diff
@@ -20,16 +20,21 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use super::{Frame, MessageError, MessageFlags};
+use super::{MessageError, MessageFlags};
 use crate::{
     consts::ENVELOPE_VERSION,
     types::{CommsPublicKey, CommsSecretKey},
     utils::signature,
 };
+use rand::rngs::OsRng;
 use serde::{Deserialize, Serialize};
 use std::convert::TryInto;
 use tari_utilities::{message_format::MessageFormat, ByteArray};
 
+// Re-export protos
+pub use crate::proto::envelope::*;
```

Comment:
I think maybe an explicit export of the Envelope struct, this star import makes it hard to tell where the Evelope struct is that belongs to the impl below.

Reply:
Though I agree, I wanted to make it so that there's no work to be done if a new message is added to a proto file - it would just be available when added (after build.rs is run). Most editors allow you to 'go to definition'.

Reply:
Really not a big deal but I have to say that working with the proto files can be complicated and tricky. The go to definition doesn't always work and for the future people I really feel we should do as much as possible to make it easy to parse.

Reply:
In the comms crate, it does work because the files are included directly.

Reply:
Hopefully someone (maybe me) can update the code in the other crates to work like this as it really makes working with them the same as any other struct